### PR TITLE
[1/3] 3D Composability - move fsdp tests

### PR DESCRIPTION
**Pull request by mori360**
pytorch (fsdp, tp, pp) -> pytorch (composable)
Move (fsdp, tp, pp) tests under pytorch into a composable folder

**FSDP:
test/distributed/_composable/fsdp/test_fully_shard_trainin.py
-TestFullyShard2DTraining**
DP:
test/distributed/tensor/parallel/test_ddp_2d_parallel.py
TP:
test/distributed/tensor/parallel/test_fsdp_2d_parallel.py
PP:
test/distributed/pipelining/test_composability.py

=>
**distributed/_composable/test_composability/test_2d_composability.py**
distributed/_composable/test_composability/test_pp_composability.py

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #131343
* #129802
* #129801
* __->__ #129800

cc @XilunWu @H-Huang @awgu @kwen2501 @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o

### CODE REVIEW

File: .ci/pytorch/multigpu-test.sh [hunk 1]
```diff
@@ -54,6 +54,9 @@ time python test/run_test.py --verbose -i distributed/_composable/fsdp/test_full
 # Pipelining composability tests
 time python test/run_test.py --verbose -i distributed/pipelining/test_composability.py
 
+# 3D composability tests
```
Comment (awgu) [inline .ci/pytorch/multigpu-test.sh:57]: If this is going to include other tests later, then maybe say ND:
```suggestion
# ND composability tests
```
Otherwise, it should say 2D for now.

**Review by awgu — COMMENTED** [review]
If any of the following tests moved files, then you need to adjust this:
https://github.com/pytorch/pytorch/blob/55dc82bef92c66638d726431843f6137ccdc9955/.ci/pytorch/test.sh#L321-L326

The tests shown above run as part of the `ciflow/inductor` label. The tests you added here in `multigpu-test.sh` run with the `ciflow/periodic` label. 

@weifengpy is there a reason we want to move these tests to periodic instead of keeping them in inductor tests?

**Comment by awgu** [comment]
Whether these tests are in `ciflow/inductor` or `ciflow/periodic`, I think we should run those tests on this PR before landing. 

For example:
Screenshot 2024-07-12 at 1 25 16 PM

Note that `ciflow/periodic` has to be manually added every time. For `ciflow/inductor`, there are some rules today on if X file is changed, then it auto adds `ciflow/inductor`, which is partially why we may prefer to keep the tests there.

**Comment by weifengpy** [comment]
> is there a reason we want to move these tests to periodic instead of keeping them in inductor tests?

TL;DR is we can keep FSDP tests in `pytorch/.ci/pytorch/test.sh` and just update the filepath. I am guessing that @mori360 made the change by following example PRs around `multigpu-test.sh`. so no particular reasons for switching

BTW, it's good to learn that `multigpu-test.sh` is associated with `ciflow/periodic` label. It's new to me

File: .ci/pytorch/test.sh [hunk 2]
```diff
@@ -329,6 +328,7 @@ test_inductor_distributed() {
   python test/run_test.py -i distributed/_composable/fsdp/test_fully_shard_mixed_precision.py -k test_reduce_dtype --verbose
   python test/run_test.py -i distributed/_composable/fsdp/test_fully_shard_clip_grad_norm_.py -k test_clip_grad_norm_2d --verbose
   python test/run_test.py -i distributed/fsdp/test_fsdp_tp_integration.py -k test_fsdp_tp_integration --verbose
+  python test/run_test.py -i distributed/_composable/test_composability/test_2d_composability.py -k test_train_parity_2d_mlp --verbose
```
Comment (awgu) [inline .ci/pytorch/test.sh:331]: curious: why is this moved?

also, is `test_2d_composability` meant to be both in `test_inductor_distributed` (`test.sh`) and `multigpu-test.sh`?

Comment (mori360) [reply .ci/pytorch/test.sh:331]: will recover it

**Review by awgu — APPROVED** [review]
From discussion: consolidating ND composability tests in `multigpu-test.sh`.

Assuming that otherwise just code motion of unit tests.
